File: c++/arrays/arrays.cpp
#include <iostream>

int main()
{
  int numbers[3];

  std::cout << "Not initialised, before assignments:" << std::endl << std::endl;

  std::cout << "numbers[0]: " << numbers[0] << std::endl;
  std::cout << "numbers[1]: " << numbers[1] << std::endl;
  std::cout << "numbers[2]: " << numbers[2] << std::endl << std::endl;

  numbers[0] = 2;
  numbers[1] = 4;
  numbers[2] = 6;

  std::cout << "After assignments:" << std::endl << std::endl;

  std::cout << "numbers[0]: " << numbers[0] << std::endl;
  std::cout << "numbers[1]: " << numbers[1] << std::endl;
  std::cout << "numbers[2]: " << numbers[2] << std::endl << std::endl;

  double values[] = {9.5, 8.5, 7.5};

  std::cout << "Initialised:" << std::endl << std::endl;

  std::cout << "values[0]: " << values[0] << std::endl;
  std::cout << "values[1]: " << values[1] << std::endl;
  std::cout << "values[2]: " << values[2] << std::endl << std::endl;

  std::cout << "Not legally allocated array cell:" << std::endl << std::endl;

  std::cout << "values[3]: " << values[3] << std::endl;
}

File: pascal/arrays/arrays.pas
program Arrays;

var
  numbers: array[0..2] of integer;
  values: array[1..3] of real = (9.5, 8.5, 7.5);

begin
  // Array elements must be initialised before access.
  numbers[0] := 2;
  numbers[1] := 4;
  numbers[2] := 6;

  writeln('After assignments:');
  writeln();
  writeln('numbers[0]: ', numbers[0]);
  writeln('numbers[1]: ', numbers[1]);
  writeln('numbers[2]: ', numbers[2]);
  writeln();

  writeln('Initialised:');
  writeln();
  writeln('values[1]: ', values[1]);
  writeln('values[2]: ', values[2]);
  writeln('values[3]: ', values[3]);
  writeln();
end.
